permission_handler_android/android/src/main/java/com/baseflow/permissionhandler/PermissionManager.java

```diff
@@ -262,44 +319,50 @@ void requestPermissions(
                 continue;
             }
 
+            // Request special permissions.
             if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && permission == PermissionConstants.PERMISSION_GROUP_IGNORE_BATTERY_OPTIMIZATIONS) {
                 executeIntent(
                         Settings.ACTION_REQUEST_IGNORE_BATTERY_OPTIMIZATIONS,
                         PermissionConstants.PERMISSION_CODE_IGNORE_BATTERY_OPTIMIZATIONS);
+                pendingRequestCount++;
```

Comment:
Won't it be better to place this line inside the `executeIntent` method, that way we don't have to repeat it and would also ensure it is incremented when new permissions are added.

Reply:
I'm a bit hesitant to do that, as the method name does not imply any side-effects. Also, there are two methods that would need to do this (`executeSimpleIntent` as well). Can we think of a cleaner way to handle this? Maybe we could refactor the methods to a single method called `launchSpecialPermission`. We could add documentation about the special special case for `NOTIFICATION_POLICY_ACCESS`. The bookkeeping makes a little more sense in such a method, as the name implies that it considers launching a permission.

Reply:
Agreed, lets go for the refactor version. I think it will be worth as it potentially will prevent bugs when new permissions are added.